src/persist-client/src/internal/compact.rs

```diff
@@ -197,11 +197,11 @@ where
         // were just written, but it does result in non-trivial blob traffic
         // (especially in aggregate). This heuristic is something we'll need to
         // tune over time.
-        let should_compact = req.inputs.len() >= self.cfg.compaction_heuristic_min_inputs
+        let should_compact = req.inputs.len() >= self.cfg.dynamic.compaction_heuristic_min_inputs()
```

Comment:
how were we thinking of setting these in LD? are we able to atomically update multiple values, or is it possible we'd read these 3 knobs with arbitrary skew

Reply:
great q, dunno! if we can handle the latter, certainly seems less brittle to bugs in the config propagation (and we could probably relax these SeqCsts once we thought about Ordering a bit, whereas atomic updates would require going _more_ expensive and having something like a mutex around all the dynamic bits). i'd love it if we could make these functions something that are safe to call in hot-ish loops (fewer footguns that way)

Reply:
yeah I like the idea of relaxing the orderings. if we can't get tight guarantees from LD anyway, we might as well minimize the cost of reading the values in mem

Reply:
I don't think it would be possible to atomically update multiple parameters in LD, at least not if those are represented as separate flags. What we could do is represent them as a single text/json flag in LD, do the decoding in environmentd, and ensure that our code always propagates them as one.

That would be less ergonomic from an LD-user perspective and more of a hassle to implement, so if persist could be taught to handle skewed individual updates instead, that would definitely be nicer!

Reply:
Sounds like we have our answer!

Reply:
Documented this as part of the new apply method.